src/middlewared/middlewared/plugins/catalogs_linux/update.py

```diff
@@ -173,10 +172,12 @@ async def do_create(self, job, data):
         job.set_progress(60, 'Completed Validation')
 
         await self.middleware.call('datastore.insert', self._config.datastore, data)
+        job.set_progress(70, f'Successfully added {data["label"]!r} catalog')
 
-        asyncio.ensure_future(self.middleware.call('catalog.sync', data['label']))
+        job.set_progress(80, f'Syncing {data["label"]} catalog')
+        await self.middleware.call('catalog.sync', data['label'])
```

Comment:
This method seems to go to the internet to update git repository information. We create a system alert `CatalogSyncFailed` if any exception occurs but I'm curious if we're going to run into a situation where we have a `CallTimeout` since it gets raised after 60 seconds by default. A slow connection or some weird DNS issue could easily lapse 60 seconds. Thoughts on that situation or do you just middlewared to raise it?

Reply:
So this is fixed in https://github.com/truenas/middleware/pull/7063 where this has been converted to a job, but this change needs to go in first before https://github.com/truenas/middleware/pull/7063 can be merged

Reply:
Also internally we don't have a `CallTimeout`, so if that method takes potentially more then a minute, it wouldn't error out.